Upgrade `syn`, Migrate `SerializeHierarchy` to `PathSerialize`, `PathDeserialize`, `PathIntrospect` (#987)
HULKs/hulk: docs/setup/upload.md

# Uploading HULK

> This section assumes, you have set up your development environment, and flashed a Yocto image to the NAO.

Reviewer: I think this sentence needs neither comma.
```suggestion
This section assumes that you have set up your development environment and flashed a Yocto image to the NAO.
```

Author: done

> Uploading the software to a NAO is task of Pepsi.
> Use `pepsi --help` to inspect the respective commands and feature flags.
> To upload the robotics software to the NAO, just run:

Reviewer: ```suggestion
Uploading the robotics software to a NAO is done with pepsi.
See `pepsi upload --help` for details.
To simply upload the software to the NAO, run:
```

Author: done

```sh
pepsi upload <NAO>
```

Pepsi takes care of downloading and installing the SDK, calling `cargo` to trigger compilation, and uploading the software to the robot.
When successful, you are presented with a green tick in your shell, and the robot is showing rotating rainbow eyes.

## SDK Management

Pepsi automatically checks for the latest SDK configured in the repository and installs it if necessary.

> To more specifically manage your SDK installation, we provide the `sdk` subcommand for Pepsi.
> Have a look at
> 
> ```sh
> pepsi sdk --help
> ```

Reviewer: ```suggestion
To manually manage your SDK installation, use the `sdk` subcommand.

```sh
pepsi sdk install --help
```
```

Author: done
